search/ranker.cpp

```diff
@@ -391,96 +347,148 @@ Result Ranker::MakeResult(PreResult2 const & r) const
   return res;
 }
 
-void Ranker::MakeResultHighlight(Result & res) const
+void Ranker::SuggestStrings()
 {
-  using TIter = buffer_vector<strings::UniString, 32>::const_iterator;
-  using TCombinedIter = CombinedIter<TIter, strings::UniString>;
+  if (m_params.m_prefix.empty() || !m_params.m_suggestsEnabled)
+    return;
 
-  TCombinedIter beg(m_params.m_tokens.begin(), m_params.m_tokens.end(),
-                    m_params.m_prefix.empty() ? 0 : &m_params.m_prefix);
-  TCombinedIter end(m_params.m_tokens.end(), m_params.m_tokens.end(), 0);
-  auto assignHighlightRange = [&](pair<uint16_t, uint16_t> const & range)
-  {
-    res.AddHighlightRange(range);
-  };
+  string prologue;
+  GetStringPrefix(m_params.m_query, prologue);
```

Comment:
Why not to get prefix from params?

Reply:
The new name is DropLastToken, so it does not "get" the prefix but rather chops it off.